mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusLifeCycleTest.java

```diff
@@ -0,0 +1,262 @@
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+class RabbitMQEventBusLifeCycleTest {
+
+    static class RabbitMQEventBusDeclarationExtension implements BeforeEachCallback, AfterEachCallback {
+
+        private EventSerializer eventSerializer;
+        private RabbitMQConnectionFactory connectionFactory;
+        private RabbitMQManagementAPI rabbitMQManagement;
+        private RoutingKeyConverter routingKeyConverter;
+
+        @Override
+        public void beforeEach(ExtensionContext context) throws Exception {
+            rabbitMQExtension.beforeEach(context);
+
+            rabbitMQManagement = rabbitMQExtension.managementAPI();
+            connectionFactory = rabbitMQExtension.getConnectionFactory();
+
+            TestId.Factory mailboxIdFactory = new TestId.Factory();
+            routingKeyConverter = RoutingKeyConverter.forFactories(new MailboxIdRegistrationKey.Factory(mailboxIdFactory));
+            eventSerializer = new EventSerializer(mailboxIdFactory, new TestMessageId.Factory());
+        }
+
+        @Override
+        public void afterEach(ExtensionContext context) throws Exception {
+            rabbitMQExtension.afterEach(context);
+        }
+
+        private RabbitMQEventBus newEventBus() {
+            return new RabbitMQEventBus(connectionFactory, eventSerializer, routingKeyConverter);
+        }
+    }
+
+    private static class GroupA extends Group {}
+
+    static RabbitMQExtension rabbitMQExtension = new RabbitMQExtension();
+    @RegisterExtension
+    static RabbitMQEventBusDeclarationExtension testExtension = new RabbitMQEventBusDeclarationExtension();
+
+    @BeforeAll
+    static void setUp() {
+        rabbitMQExtension.beforeAll(null);
+    }
+
+    @AfterAll
+    static void tearDown() {
+        rabbitMQExtension.afterAll(null);
+    }
+
+    @Nested
+    class SingleEventBus {
+
+        private RabbitMQEventBus eventBus;
+
+        @BeforeEach
+        void setUp() {
+            eventBus = testExtension.newEventBus();
+        }
+
+        @AfterEach
+        void tearDown() {
+            eventBus.stop();
+        }
+
+        @Test
+        void startShouldCreateEventExchange() {
+            eventBus.start();
+            assertThat(testExtension.rabbitMQManagement.listExchanges())
+                .filteredOn(exchange -> exchange.getName().equals(MAILBOX_EVENT_EXCHANGE_NAME))
+                .hasOnlyOneElementSatisfying(exchange -> {
+                    assertThat(exchange.isDurable()).isTrue();
+                    assertThat(exchange.getType()).isEqualTo(DIRECT_EXCHANGE);
+                });
+        }
+
+        @Disabled("queue is declared with durable = true but management API always return durable = false. Suspecting a bug" +
+            " from rabbitmq driver")
```

Comment:
We are the one writting that so called "driver". Wouldn't it be a bug of the RabbitMQ Management API?

(Wouldn't it deserve a ticket?)

Reply:
issue is reported https://github.com/reactor/reactor-rabbitmq/issues/44